website/for_teachers.py

```diff
@@ -1123,26 +1145,71 @@ def parse_preview_adventure(self):
             return gettext("something_went_wrong_keyword_parsing"), 400
         return {"code": code}, 200
 
-    @route("/create-adventure", methods=["POST"])
+    def add_adventure_to_class_level(self, user, class_id, adventure_id, level, remove):
+        Class = self.db.get_class(class_id)
+        if not Class or (not utils.can_edit_class(user, Class) and not is_admin(user)):
+            return utils.error_page(error=404, ui_message=gettext("no_such_class"))
+
+        customizations, adventures, adventure_names, _, _ = self.get_class_info(
+            user, class_id)
+
+        teacher_adventures = list(self.db.get_teacher_adventures(user["username"]))
+        second_teacher_adventures = self.db.get_second_teacher_adventures([Class], user["username"])
+        teacher_adventures += second_teacher_adventures
+        is_teacher_adventure = self.is_adventure_from_teacher(adventure_id, teacher_adventures)
+
+        if not remove and any(adventure['name'] == adventure_id
+                              for adventure in customizations['sorted_adventures'][level]):
+            return
+        if not remove:
+            customizations['sorted_adventures'][level].append(
+                {'name': adventure_id, 'from_teacher': is_teacher_adventure})
+        else:
+            customizations['sorted_adventures'][level].remove(
+                {'name': adventure_id, 'from_teacher': is_teacher_adventure})
+        sorted_adventure = SortedAdventure(short_name=adventure_id,
+                                           long_name=adventure_names[adventure_id],
+                                           is_teacher_adventure=is_teacher_adventure,
+                                           is_command_adventure=adventure_id in hedy_content.KEYWORDS_ADVENTURES)
+
+        if not remove:
+            adventures[int(level)].append(sorted_adventure)
+        else:
+            adventures[int(level)].remove(sorted_adventure)
+        self.reorder_adventures(adventures[int(level)], from_sorted_adv_class=True)
```

Comment:
why are you doing this step actually? Any specific reason? Because i don't think you need it!

Reply:
You told me to use `self.reorder_adventures(adventures[int(level)], from_sorted_adv_class=True)`:

> * whenever you add an adventure to class' levels, use the function i created that makes sure the order of puzzles and quizes are always at last; https://github.com/hedyorg/hedy/blob/main/website/for_teachers.py#L398

Reply:
Or do you mean `adventures[int(level)].remove(sorted_adventure)`?

Reply:
i mean the adventures variable doesn't need to be updated it. This is because it's updated automatically from the customizations["sorted_adventures"] in the get_class_info function. Thus, you don't need to update it here!